62.CAD/Hatch.cpp

```diff
@@ -0,0 +1,782 @@
+	return position;
+}
+
+// https://pomax.github.io/bezierinfo/#pointvectors
+float64_t2 Hatch::QuadraticBezier::tangent(double t) const
+{
+	// TODO: figure out a tangent algorithm for when this becomes A, B, C
+	auto derivativeOrder1First = 2.0 * (p[1] - p[0]);
+	auto derivativeOrder1Second = 2.0 * (p[2] - p[1]);
+	auto tangent = (1.0 - t) * derivativeOrder1First + t * derivativeOrder1Second;
+	auto len = sqrt(tangent.x * tangent.x + tangent.y * tangent.y);
+	return tangent / len;
+}
+
+Hatch::QuadraticBezier Hatch::QuadraticBezier::splitCurveTakeLower(double t) const
+{
+	QuadraticBezier outputCurve;
+	outputCurve.p[0] = p[0];
+	outputCurve.p[1] = (1 - t) * p[0] + t * p[1];
+	outputCurve.p[2] = (1 - t) * ((1 - t) * p[0] + t * p[1]) + t * ((1 - t) * p[1] + t * p[2]);
+	assert(outputCurve.evaluateBezier(0.0)[(int)SelectedMajorAxis] <= outputCurve.evaluateBezier(1.0)[(int)SelectedMajorAxis]);
+
+	return outputCurve;
+}
+
+Hatch::QuadraticBezier Hatch::QuadraticBezier::splitCurveTakeUpper(double t) const
+{
+	QuadraticBezier outputCurve;
+	outputCurve.p[0] = (1 - t) * ((1 - t) * p[0] + t * p[1]) + t * ((1 - t) * p[1] + t * p[2]);
+	outputCurve.p[1] = (1 - t) * p[1] + t * p[2];
+	outputCurve.p[2] = p[2]; 
+	assert(outputCurve.evaluateBezier(0.0)[(int)SelectedMajorAxis] <= outputCurve.evaluateBezier(1.0)[(int)SelectedMajorAxis]);
+
+	return outputCurve;
+}
+
+bool Hatch::QuadraticBezier::splitIntoMajorMonotonicSegments(std::array<Hatch::QuadraticBezier, 2>& out) const
+{
+	// Getting derivatives for our quadratic bezier
+	auto major = (uint)SelectedMajorAxis;
+	auto a = 2.0 * p[1][major] - p[0][major];
+	auto b = 2.0 * p[2][major] - p[1][major];
+
+	// Finding roots for the quadratic bezier derivatives (a straight line)
+	auto rcp = 1.0 / (b - a);
+	auto t = -a * rcp;
+	if (isinf(rcp) || t <= 0.0 || t >= 1.0) return true;
+	out = { splitCurveTakeLower(t), splitCurveTakeUpper(t) };
+	return false;
+}
+
+// https://pomax.github.io/bezierinfo/#boundingbox
+std::pair<float64_t2, float64_t2> Hatch::QuadraticBezier::getBezierBoundingBoxMinor() const
+{
+	auto minor = (uint)SelectedMinorAxis;
+	double A = p[0][minor] - 2.0 * p[1][minor] + p[2][minor];
+	double B = 2.0 * (p[1][minor] - p[0][minor]);
+
+	const int searchTSize = 3;
+	double searchT[searchTSize];
+	searchT[0] = 0.0;
+	searchT[1] = 1.0;
+	searchT[2] = -B / (2 * A);
+
+	float64_t2 min = float64_t2(std::numeric_limits<double>::infinity());
+	float64_t2 max = float64_t2(-std::numeric_limits<double>::infinity());
+
+	for (uint32_t i = 0; i < searchTSize; i++)
+	{
+		double t = searchT[i];
+		if (t < 0.0 || t > 1.0 || isnan(t))
+			continue;
+		float64_t2 value = evaluateBezier(t);
+		min = float64_t2(std::min(min.x, value.x), std::min(min.y, value.y));
+		max = float64_t2(std::max(max.x, value.x), std::max(max.y, value.y));
+	}
+
+	return std::pair<float64_t2, float64_t2>(min, max);
+}
+
+bool Hatch::QuadraticBezier::isLineSegment() const
+{
+	float64_t2 A = p[0] - 2.0 * p[1] + p[2];
```

Comment:
instead of getting A by manual calculation. now that your quadratic bezier in `nbl::hlsl::QuadraticBezier`
do a `nbl::hlsl::Quadratic::constructFromBezier()` and do it like @Przemog1 did it here 
https://github.com/Devsh-Graphics-Programming/Nabla/blob/4d6e017f97b5782f78326666267a789be0790636/include/nbl/builtin/hlsl/shapes/beziers.hlsl#L297